packages/navigator/src/browser/navigator-filter.ts

```diff
@@ -58,7 +63,22 @@ export class FileNavigatorFilter {
     }
 
     protected createFilterPredicate(exclusions: FileNavigatorFilter.Exclusions): FileNavigatorFilter.Predicate {
-        return new FileNavigatorFilterPredicate(exclusions);
+        return new FileNavigatorFilterPredicate(this.interceptExclusions(exclusions));
+    }
+
+    hiddenFilesVisible(): boolean {
+        return this.showHiddenFilesFlag;
+    }
+
+    toggleHiddenFiles(visible: boolean): void {
+        this.showHiddenFilesFlag = visible;
+        this.filterPredicate = this.createFilterPredicate(this.exclusions || {});
+        this.fireFilterChanged();
+    }
+
+    private interceptExclusions(exclusions: FileNavigatorFilter.Exclusions): FileNavigatorFilter.Exclusions {
+        const origExclusions = Object.assign({}, exclusions);
+        return Object.assign(origExclusions, {"**/.*": this.showHiddenFilesFlag});
```

Comment:
Cannot we simplify with:
```typescript
return {
  ...exclusions,
  '**/.*': this.showHiddenFilesFlag
};
```

Reply:
Thanks for the hint. Didn't know about that trick. Fixed.